### PR TITLE
Update backstage package versions

### PR DESCRIPTION
**Pull request by tylerd-canva**
Would like to get this released ASAP as it is blocking my own org's backstage upgrade due to the incompatibility in the scaffolder packages: https://github.com/backstage/backstage/blob/master/plugins/scaffolder-backend/CHANGELOG.md#1270

#### :heavy_check_mark: Checklist

- [ ] Added tests for new functionality and regression tests for bug fixes
- [ ] Added changeset (run `yarn changeset` in the root)
- [ ] Screenshots of before and after attached (for UI changes)
- [ ] Added or updated documentation (if applicable)

### CODE REVIEW

**Comment by tylerd-canva** [comment]
looking into the test failures

**Comment by tylerd-canva** [comment]
Okay, I've got to give up for now. I have no idea what's going on. Somehow this is getting called with no `ref`: https://github.com/backstage/backstage/blob/6db849e650901cf11e7d5b2758b518381e118dc4/packages/frontend-plugin-api/src/extensions/createComponentExtension.tsx#L36

hence the stacktrace:

```
      at createComponentExtension (../../../../node_modules/@backstage/frontend-plugin-api/dist/extensions/createComponentExtension.esm.js:8:11)
      at Object.<anonymous> (../../../../node_modules/@backstage/plugin-app/dist/extensions/components.esm.js:39:67)
      at Object.<anonymous> (../../../../node_modules/@backstage/plugin-app/dist/plugin.esm.js:24:24)
      at Object.<anonymous> (../../../../node_modules/@backstage/plugin-app/dist/index.esm.js:11:20)
      at Object.<anonymous> (../../../../node_modules/@backstage/frontend-test-utils/dist/app/renderInTestApp.esm.js:17:59)
      at Object.<anonymous> (../../../../node_modules/@backstage/frontend-test-utils/dist/index.esm.js:59:29)
      at Object.<anonymous> (alpha/entityContent.test.tsx:7:28)
```

I even hacked in a debug message:

```
[...]
function createComponentExtension(options) {
  if (options.ref === undefined) {
    throw new Error("Component extensions must specify a ref: "+ JSON.stringify(options));
  }
  return createExtension({
[...]
```

and I get `Component extensions must specify a ref: {"loader":{}}`.

File: package.json
Comment (tylerd-canva) [inline package.json:59]: this is being added automatically because I'm using corepack locally. It shouldn't hurt anything and in theory gives you more consistency across contributors.

File: package.json
Comment (tylerd-canva) [inline package.json:44]: moved this from normal deps to devDeps, everything should be in devDeps in the root of a monorepo AFAIK.

File: packages/backend/package.json
Comment (tylerd-canva) [inline packages/backend/package.json:10]: this is redundant and out of date -- you only need to specify this in the monorepo root package.json

File: plugins/frontend/backstage-plugin-github-insights/src/components/Widgets/ReadMeCard/ReadmeCard.test.tsx
Comment (tylerd-canva) [inline plugins/frontend/backstage-plugin-github-insights/src/components/Widgets/ReadMeCard/ReadmeCard.test.tsx:1]: I've got no idea why this test was failing inconsistently for me, but this seems to help when I run it locally.

**Comment by tylerd-canva** [comment]
All the tests are passing locally for me, now.

**Comment by tylerd-canva** [comment]
Is anyone able to look at this? Happy to rebase it after getting approval, just don't want to keep rebasing it over and over.

**Comment by tylerd-canva** [comment]
The Backstage 1.34 release cycle has begun. If I can't get these Roadie packages updated soon I will need to migrate away from them or fork them in order to block our own upgrade. Please let me know if there's any issues with this PR so I can get it fixed up, rebased, and merged ASAP. Thank you!

**Comment by Xantier** [comment]
Thanks @tylerd-canva , 

I should have some bandwidth tomorrow to take a closer look at this

**Comment by Xantier** [comment]
This looks good to go. Can you rebase and we will get this merged

**Comment by tylerd-canva** [comment]
@Xantier Thank you!!! Done!

**Review by Xantier — APPROVED** [review]

**Pull request merged**
